models/CloudServiceProviders.ts

```diff
@@ -0,0 +1,26 @@
+export enum CloudServiceProvider {
+  CSP_A = "CSP_A",
+  CSP_B = "CSP_B",
+  CSP_C = "CSP_C",
+  CSP_D = "CSP_D",
+}
+
+export enum CloudServiceProviderUri {
+  A = "www.cspA.com/v1",
+  B = "www.cspB.com/v1",
+  C = "www.cspC.com/v1",
+  D = "www.cspD.com/v1",
+}
+
+export enum Network {
+  NETWORK_1 = "NETWORK_1",
+  NETWORK_2 = "NETWORK_2",
+  NETWORK_3 = "NETWORK_3",
+}
+
+export const CSP_URI = {
+  [CloudServiceProvider.CSP_A]: CloudServiceProviderUri.A,
+  [CloudServiceProvider.CSP_B]: CloudServiceProviderUri.B,
+  [CloudServiceProvider.CSP_C]: CloudServiceProviderUri.C,
+  [CloudServiceProvider.CSP_D]: CloudServiceProviderUri.D,
+};
```

Comment:
Would it be better to do:

```ts
export interface ICloudServiceProvider {
  name: string;
  uri: string;
  networks: Network[];
}

export class CloudServiceProvider implements ICloudServiceProvider {
  // IDK if these modifiers are valid together but it does need to be static and public; readonly may
  // be implied by that
  public static readonly CSP_A = new CloudServiceProvider("CSP_A", "www.cspA.com/v1", [Network.A, Network.B]);
  public static readonly CSP_B = new CloudServiceProvider("CSP_B", "www.cspB.com/v1", []);
  public static readonly CSP_C = new CloudServiceProvider("CSP_C", "www.cspC.com/v1", [Network.A]);
  public static readonly CSP_D = new CloudServiceProvider("CSP_D", "www.cspD.com/v1", [Network.C]);

  private constructor(name: string, uri: string, networks: Network[]) {
    this.name = name;
    this.uri = uri;
    this.networks = [...networks];
  }
}
```

This way you always have CSP A associated to its name, URI (which let's be careful because that's not a URI per spec), and networks and you can't ever accidentally do something like:

```
POST ${CloudServiceProviderUri.B}

{
  "CSP": "${CloudServiceProvider.A}"
}

Reply:
That is definitely better and avoid some accidents. Thanks for the suggestion! @kylelaker-ccpo Can you expand a little on the URI portion not being a URI per the spec?

Reply:
@juliusfitzhugh-ccpo I think you've got it right.  My interpretation is that he's saying `www.cspD.com/v1` is not, [per spec](https://datatracker.ietf.org/doc/html/rfc3986), a URI, but your example `https://www.cspA.com/` would be.

Reply:
Correct! We need to specify a scheme and `https://` would correct. In general, we probably want to _assume_ and _require_ HTTPS. So everything you currently have but with `https://` prepended would be fine!

Reply:
I will update it with `https://`, thanks for clearing that up.